frontend/awx/administration/instance-groups/hooks/useInstanceGroupActions.tsx

```diff
@@ -0,0 +1,119 @@
+import { EditIcon, PlusCircleIcon, TrashIcon } from '@patternfly/react-icons';
+import { AwxRoute } from '../../../AwxRoutes';
+import { useDeleteInstanceGroups } from './useDeleteInstanceGroups';
+import { ActionsResponse, OptionsResponse } from '../../../interfaces/OptionsResponse';
+import { awxAPI } from '../../../api/awx-utils';
+import { useOptions } from '../../../../common/crud/useOptions';
+
+export function useDisableCreateInstanceGroup() {
+  const { data } = useOptions<OptionsResponse<ActionsResponse>>(awxAPI`/instance_groups/`);
+  const { t } = useTranslation();
+  return useMemo(
+    () =>
+      data && data.actions && data.actions['POST']
+        ? undefined
+        : t(
+            'You do not have permission to create an instance group. Please contact your organization administrator if there is an issue with your access.'
+          ),
+    [data, t]
+  );
+}
+
+export function useInstanceGroupToolbarActions(
+  onComplete: (instanceGroups: InstanceGroup[]) => void
+) {
+  const { t } = useTranslation();
+  const pageNavigate = usePageNavigate();
+  const deleteInstanceGroups = useDeleteInstanceGroups(onComplete);
+  const isCreateActionDisabled = useDisableCreateInstanceGroup();
+
+  return useMemo<IPageAction<InstanceGroup>[]>(
+    () => [
+      {
+        type: PageActionType.Dropdown,
+        icon: PlusCircleIcon,
+        variant: ButtonVariant.primary,
+        isPinned: true,
+        selection: PageActionSelection.None,
+        label: t('Create group'),
+        isDisabled: isCreateActionDisabled,
+        actions: [
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.None,
+            label: t('Create container group'),
+            onClick: () => pageNavigate(AwxRoute.CreateInstanceGroup),
+          },
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.None,
+            label: t('Create instance group'),
+            onClick: () => pageNavigate(AwxRoute.CreateInstanceGroup),
+          },
+        ],
+      },
+      { type: PageActionType.Seperator },
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Multiple,
+        icon: TrashIcon,
+        label: t('Delete selected instance groups'),
+        onClick: deleteInstanceGroups,
+        isDanger: true,
+      },
+    ],
+    [deleteInstanceGroups, isCreateActionDisabled, pageNavigate, t]
+  );
+}
+
+export function useInstanceGroupRowActions(onComplete: (instanceGroups: InstanceGroup[]) => void) {
+  const { t } = useTranslation();
+  const pageNavigate = usePageNavigate();
+  const deleteInstanceGroups = useDeleteInstanceGroups(onComplete);
+
+  return useMemo<IPageAction<InstanceGroup>[]>(
+    () => [
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Single,
+        icon: EditIcon,
+        isPinned: true,
+        label: t('Edit instance group'),
```

Comment:
On these row actions we should distinguish between instance groups and container groups.

Reply:
That's a good point! It will require an update to the framework component to accept a function as label.

Reply:
Updated 👍 

[PS: The attempt at having the PageAction framework component accept a function for `label` introduced several Typescript issues across the project. This would be a larger undertaking and if we have more such use cases it would make sense to add it. For now, I was able to get around that by adding separate actions for instance vs. container groups.]

Reply:
Updated 👍 

[PS: The attempt at having the PageAction framework component accept a function for `label` introduced several Typescript issues across the project. This would be a larger undertaking and if we have more such use cases it would make sense to add it. For now, I was able to get around that by adding separate actions for instance vs. container groups.]